### PR TITLE
Task #210890 ,Task #210891 All Cohort wise academic_year and program_id added in edit_requests module and user module and acitivies and beneficiaries module

### PR DESCRIPTION
**Pull request by Tusharmahajan12**
https://tracker.tekdi.net/issues/210890
https://tracker.tekdi.net/issues/210891

# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **New Features**
  - Added `TaxonomyModule` for improved taxonomy management.
  - Introduced new API endpoints for retrieving program and academic year details.

- **Enhancements**
  - Improved search functionality by refining parameter handling.
  - Streamlined user update process with additional request parameter integration.
  - Enhanced edit request services with support for additional context parameters.

- **Bug Fixes**
  - Corrected default assignment of program identifiers in beneficiary services.
  - Adjusted facilitator service to correctly utilize user context for data retrieval.

- **Refactor**
  - Reordered modules for better organization and maintainability.
  - Updated services to use request properties for program and academic year identifiers.

- **Removals**
  - Removed unused route `/beneficiaries/admin/reassign` from `BeneficiariesModule`.

- **Documentation**
  - No visible changes to end-user documentation in this release.

### CODE REVIEW

File: src/src/user/user.service.ts [hunk 1]
```diff
@@ -1357,6 +1357,7 @@ export class UserService {
 			}
 		  }
 		  `;
+console.log("vccc",validation_query);
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/src/user/user.service.ts:1360]: @Tusharmahajan12 remove comment

File: src/src/user/user.module.ts [hunk 2]
```diff
@@ -1,5 +1,8 @@
 import { MiddlewareConsumer, Module, NestModule } from '@nestjs/common';
+<<<<<<< HEAD
```
Comment (manojLondhe) [inline src/src/user/user.module.ts:2]: fix conflicts @Tusharmahajan12

File: src/src/user/user.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tusharmahajan12) [reply src/src/user/user.service.ts]: removed console

File: src/src/user/user.module.ts [hunk 3]
```diff
@@ -1,5 +1,8 @@
...
+=======
 import { CohortMiddleware } from 'src/common/middlewares/cohort.middleware';
+>>>>>>> upstream/release-1.7.0
```
[the remark this answers is not in the document]
Comment (Tusharmahajan12) [reply src/src/user/user.module.ts]: solved confilict

File: src/src/user/user.module.ts (hunk 2 above)
Comment (Tusharmahajan12) [reply src/src/user/user.module.ts]: solved conflicts

File: src/src/modules/edit-requests/edit-requests.core.service.ts [hunk 4]
```diff
@@ -77,13 +77,13 @@ export class EditRequestCoreService {
...
-		let {
-			program_id,
-			academic_year_id,
-			edit_req_for_context,
-			edit_req_for_context_id,
-		} = body;
+	public async getEditRequestList(
+		body,
+		edit_req_by,
+		program_id,
```
Comment (manojLondhe) [inline src/src/modules/edit-requests/edit-requests.core.service.ts:83]: do we need to pass these as params?

why not get form request.mw ?

File: src/src/user/user.service.ts [hunk 5]
```diff
@@ -24,10 +24,10 @@ export class UserService {
 		private readonly keycloakService: KeycloakService,
 	) {}
 
-	public async update(userId: string, body: any, tableName: String) {
+	public async update(userId: string, body: any, tableName: String,req:any) {
 		try {
-			let program_id = body?.program_id || 1;
-			let academic_year_id = body?.academic_year_id || 1;
+			const program_id = req.mw_program_id;
+		const academic_year_id = req.mw_academic_year_id;
```
Comment (manojLondhe) [inline src/src/user/user.service.ts:30]: fix indent

File: src/src/user/user.controller.ts [hunk 6]
```diff
@@ -118,8 +118,9 @@ export class UserController {
 	public async updateUser(
 		@Param('id') id: string,
 		@Body() body: Record<string, any>,
+		@Req() req:any,
 	) {
-		return this.userService.update(id, body, 'program_faciltators');
+		return this.userService.update(id, body,req, 'program_faciltators');
```
Comment (manojLondhe) [inline src/src/user/user.controller.ts:123]: add space before and after `,`

**Pull request merged**